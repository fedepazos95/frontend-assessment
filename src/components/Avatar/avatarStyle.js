const avatarStyle = {
  fluid: {
    maxWidth: '100%',
    height: 'auto',
  },
  rounded: {
    borderRadius: '6px !important',
  },
};

export default avatarStyle;
