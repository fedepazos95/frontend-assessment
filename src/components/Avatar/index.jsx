import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import styles from './avatarStyle';

const useStyles = makeStyles(styles);

export default function Avatar({ image, alt }) {
  const classes = useStyles();

  return (
    <img
      src={image}
      alt={alt}
      className={`${classes.rounded} ${classes.fluid}`}
    />
  );
}

Avatar.defaultProps = {
  alt: '',
};

Avatar.propTypes = {
  image: PropTypes.string.isRequired,
  alt: PropTypes.string,
};
