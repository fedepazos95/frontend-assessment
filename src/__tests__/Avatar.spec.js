/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Avatar from 'components/Avatar';

configure({ adapter: new Adapter() });

describe('<Avatar /> with no props', () => {
  const container = shallow(<Avatar />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
  it('should have an image field', () => {
    expect(container.find('img').length).toEqual(1);
  });
});

describe('<Avatar /> with props', () => {
  const initialProps = {
    image: 'imageUrl',
  };
  const container = shallow(<Avatar {...initialProps} />);

  it('should have proper props for image field', () => {
    expect(container.find('img').props()).toEqual({
      alt: '',
      className: 'makeStyles-rounded-2 makeStyles-fluid-1',
      src: 'imageUrl',
    });
  });
});
