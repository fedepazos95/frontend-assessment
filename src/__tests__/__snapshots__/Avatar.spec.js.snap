// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> with no props should match the snapshot 1`] = `"<img alt=\\"\\" class=\\"undefined makeStyles-rounded-2 makeStyles-fluid-1\\"/>"`;
